x/rollup/types/adapters_test.go

```diff
@@ -0,0 +1,308 @@
+package types_test
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+
+	bfttypes "github.com/cometbft/cometbft/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	rollupv1 "github.com/polymerdao/monomer/gen/rollup/v1"
+	"github.com/polymerdao/monomer/testutils"
+	rolluptypes "github.com/polymerdao/monomer/x/rollup/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAdaptPayloadTxsToCosmosTxs(t *testing.T) {
+	t.Run("Zero txs", func(t *testing.T) {
+		cosmosTxs, err := rolluptypes.AdaptPayloadTxsToCosmosTxs([]hexutil.Bytes{})
+		require.NoError(t, err)
+		assert.Equal(t, 0, len(cosmosTxs))
+	})
+
+	src := rand.NewSource(0)
+	r := rand.New(src)
+
+	t.Run("non-zero txs without error", func(t *testing.T) {
+		testTable := []struct {
+			name              string
+			depNum, nonDepNum int
+		}{
+			{
+				name:   "DepositTx",
+				depNum: 1,
+			},
+			{
+				name:   "Multiple DepositTxs",
+				depNum: 3,
+			},
+			{
+				name:      "DepositTx + AccessListTx",
+				depNum:    1,
+				nonDepNum: 1,
+			},
+			{
+				name:      "Multiple DepositTxs + DynamicFeeTxs",
+				depNum:    3,
+				nonDepNum: 3,
+			},
+		}
+
+		interfaceRegistry := codectypes.NewInterfaceRegistry()
+		rollupv1.RegisterInterfaces(interfaceRegistry)
+		protoCodec := codec.NewProtoCodec(interfaceRegistry)
+
+		for _, tc := range testTable {
+			t.Run(tc.name, func(t *testing.T) {
+				ethTxs := make([]hexutil.Bytes, tc.depNum+tc.nonDepNum)
+				transactions := generateEthTransactions(t, tc.depNum, tc.nonDepNum)
+
+				depositTxsNum := 0
+				for i := range transactions {
+					txBinary, err := transactions[i].MarshalBinary()
+					require.NoError(t, err)
+					ethTxs[i] = txBinary
+					if transactions[i].IsDepositTx() {
+						depositTxsNum++
+					}
+				}
```

Comment:
This looks a lot like the real implementation... are we overfitting for the implementation here?